DashAI/back/dataloaders/classes/dataset_dashai.py

```diff
@@ -0,0 +1,151 @@
+                "inputs and outputs can only contain elements that exist in names"
+            )
+            # Validate that the union of inputs and outputs is equal to names
+        if set(inputs + outputs) != set(names):
+            raise ValueError("the union of inputs and outputs must be equal to names")
+
+    def cast(self, *args, **kwargs):
+        """
+        Override of the cast method to leave it in datasetdashai format
+        Args:
+            args: arguments forwarded to super.
+            **kwargs: keyword arguments forwarded to super.
+        Returns:
+            DatasetDashAI: DatasetDashAI after cast
+        """
+        ds = super().cast(*args, **kwargs)
+        return DatasetDashAI(ds._data, self.inputs_columns, self.outputs_columns)
+
+    def save_to_disk(self, dataset_path: str):
+        """
+        Override of the save to disk method to save datasetdashai info in json file
+        Args:
+            dataset_path (str): path where the datasetdashai will be stored
+        """
+        super().save_to_disk(dataset_path)
+        with open(
+            os.path.join(dataset_path, "dataset_dashai_info.json"),
+            "w",
+            encoding="utf-8",
+        ) as dashai_info_file:
+            data_dashai = {
+                "inputs_columns": self.inputs_columns,
+                "outputs_columns": self.outputs_columns,
+            }
+            json.dump(data_dashai, dashai_info_file, indent=2, sort_keys=True)
+
+    @staticmethod
+    def load_from_disk(dataset_path: str):
+        """
+        Method analogous to the Dataset load disk method, but leaving the
+        dataset in datasetdashai format
+        Args:
+            dataset_path (str): path where the datasetdashai will be stored
+        Returns:
+            DatasetDashAI: DatasetDashAI loaded from disk
+        """
+        dataset = load_from_disk(dataset_path=dataset_path)
+        with open(
+            os.path.join(dataset_path, "dataset_dashai_info.json"),
+            "r",
+            encoding="utf-8",
+        ) as dashai_info_file:
+            dataset_dashai_info = json.load(dashai_info_file)
+        inputs_columns = dataset_dashai_info["inputs_columns"]
+        outputs_columns = dataset_dashai_info["outputs_columns"]
+        return DatasetDashAI(dataset._data, inputs_columns, outputs_columns)
+
+    def change_columns_type(self, types: Dict[str, str]):
+        """
+        Method that will be implemented temporarily to change the type of some columns.
+        Args:
+            types (dict): dictionary which has as keys the names of the columns to be
+            changed and as values the new types
+        Returns:
+            DatasetDashAI: DatasetDashAI after type changes
+        """
+        if not isinstance(types, dict):
+            raise TypeError(f"types should be a dict, got {type(types)}")
+
+        # Check if columns names exists
+        columns = self.column_names
+
+        for c in types.keys():
```

Comment:
- `in` over a dict assumes that you are searching some key.
- use a more descriptive name than c, for example, `col`.

```suggestion
        for col in types:
```